axis_backend/apps/clients/services/client_service.py

```
@@ -0,0 +1,397 @@
+
+        # Check for duplicate tax ID
+        if tax_id and self.repository.find_by_tax_id(tax_id):
+            raise ValidationError(f"Client with tax ID '{tax_id}' already exists")
+
+        # Validate industry exists
+        industry = None
+        if industry_id:
+            try:
+                industry = Industry.objects.get(id=industry_id)
+            except Industry.DoesNotExist:
+                raise ValidationError(f"Industry with ID '{industry_id}' does not exist")
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)